### PR TITLE
Fix advanced editor

### PR DESCRIPTION
**Pull request by kolking**
There were more issues besides the hidden bottom part of the editor. I've tried to resolve them all in this PR, please test it well to make sure I didn't miss something.

### CODE REVIEW

**Comment by saracarl** [comment]
Review:

- Works well with full screen mode
- buttons underneath editor are moved to the right:
image
- no buttons appear once enabled.  (Possible merge conflict?)

**Comment by kolking** [comment]
@saracarl seems like a bug with buttons. I will fix, please don't merge it yet.

**Comment by benwbrum** [comment]
These changes look really good.

One problem is that the editor is no longer submitting the form to the server.

To reproduce:
* Edit a page like [this one](http://localhost:3000/benwbrum/jeremiah-white-graves-diaries/jeremiah-white-graves-diary-volume-2-book-10/transcribe/504359)
* Add text to the page
* Click the top Save button
* No POST is sent to the server, so your changes are not saved.

**Comment by kolking** [comment]
Fixed bottom toolbar buttons and the form submission

**Comment by kolking** [comment]
Should be ready to merge now.

**Comment by kolking** [comment]
@benwbrum this looks weird... Are you sure you are on the `2220-advanced-editor` branch? Could you try to clear browser cache and reload?

**Comment by kolking** [comment]
Also, does this happen with other pages / in other collections?

**Comment by benwbrum** [comment]
@kolking I've just cleared my browser cache, and visited [another page](http://localhost:3000/benwbrum/jeremiah-white-graves-diaries/jeremiah-white-graves-diary-volume-2-book-10/transcribe/504360) with the same result:

Screenshot from 2021-02-15 11-07-52

Here's the most recent entries from `git log` on my local branch:
```
commit 3ae1009b888d90693f6b860c0e51af41f7183ad0 (HEAD -> 2220-advanced-editor, origin/2220-advanced-editor)
Author: Nick Seryakov <nick.seryakov@gmail.com>
Date:   Sun Feb 14 02:43:42 2021 +0400

    Fix text wrapping and form submission

commit b5f8b50093359a7988c349c2d368810a19492811
Author: saracarl <saracarl@gmail.com>
Date:   Wed Feb 10 10:33:18 2021 -0600

    fix asset includes for removed css

commit f1f43273dd19fae42ffe21c0ab24a2854f21c00b
Author: Nick Seryakov <nick.seryakov@gmail.com>
Date:   Mon Feb 8 04:13:54 2021 +0400

    The has_title param is not required now

commit 03d0af7d0bd9ab080e5493f36bfe8373db465b4e
Author: Nick Seryakov <nick.seryakov@gmail.com>
Date:   Mon Feb 8 04:12:22 2021 +0400

    Rework transcribe page markup and styles

commit b2955b63865d2f6c3d2896ebb4fc45e68a4bb82f
Author: Nick Seryakov <nick.seryakov@gmail.com>
Date:   Mon Feb 8 04:08:51 2021 +0400

    Remove codemirror buttons stylesheet

commit 15e539e2d85fea738805f2d26dcbfe45be243253
Author: Nick Seryakov <nick.seryakov@gmail.com>
Date:   Mon Feb 8 04:08:15 2021 +0400

    Add class name for codemirror panel wrapper

commit ed71fe620c3b8259287b0d26b3f10d715ae1e605
Merge: 48f8d446 ea5e60de
Author: Sara Brumfield <saracarl@gmail.com>
Date:   Thu Feb 4 12:19:47 2021 -0600

    Merge pull request #2345 from benwbrum/1739-i18n
    
    1739 i18n
```

Did I miss something?

I wonder if the issue is the removal of the JS call to `setSize`?

**Comment by kolking** [comment]
Still weird... After your [first comment](https://github.com/benwbrum/fromthepage/pull/2349#issuecomment-777087377) (where you probably didn't have this problem, or did you?) I only made some minor [unrelated changes](https://github.com/benwbrum/fromthepage/pull/2349/commits/3ae1009b888d90693f6b860c0e51af41f7183ad0)

Regarding `setSize` - it is not required, because the layout is now based on a pure CSS using flex model. What's the browser (name / version) you are testing it now?

**Comment by benwbrum** [comment]
That's correct, we did not have this problem after the first comment.  If you'd like, I can reverse commits on my local machine to isolate the change that caused the problem.

I'm using Google Chrome on Ubuntu, Version 88.0.4324.150 (Official Build) (64-bit)

**Comment by kolking** [comment]
> If you'd like, I can reverse commits on my local machine to isolate the change that caused the problem.

Yes, please. I'm confused because I cannot reproduce this bug on my local version.

**Comment by kolking** [comment]
I'm using the same Chrome version / build. Also, could you please open Chrome dev tools and see is there any javascript errors in the console?

**Comment by benwbrum** [comment]
I have just walked through the following commits:
```
Works good:

ed71fe620c3b8259287b0d26b3f10d715ae1e605
15e539e2d85fea738805f2d26dcbfe45be243253
b2955b63865d2f6c3d2896ebb4fc45e68a4bb82f


Error with undefined local variable or method `has_title' for #<#<Class:0x00005$
03d0af7d0bd9ab080e5493f36bfe8373db465b4e

One-line editor problem:
f1f43273dd19fae42ffe21c0ab24a2854f21c00b

```

So it looks like the problem was introduced in 03d0af7d0bd9ab080e5493f36bfe8373db465b4e or f1f43273dd19fae42ffe21c0ab24a2854f21c00b


When I look in my JS console for f1f43273dd19fae42ffe21c0ab24a2854f21c00b I see no JS errors.  I'm trying to inspect the document now.

**Comment by benwbrum** [comment]
The Codemirror-wrap class appears to style one of the divs at a height of `23px`.  Eliminating that attribute seems to fix the problem in developer tools.  (See screenshots)
Screenshot from 2021-02-15 11-36-41
Screenshot from 2021-02-15 11-36-35

**Comment by kolking** [comment]
In your last screenshots, I actually see you don't have the new markup (and probably styles), it should look like this:

Снимок экрана 2021-02-15 в 21 43 09

It should be wrapped with a `<div class="cm-editor">` (in your screenshot it's just `<div>`) which will override the `CodeMirror-wrap` element's height to `height: auto !important`.

I've added the class name to the wrapper here: https://github.com/benwbrum/fromthepage/pull/2349/commits/15e539e2d85fea738805f2d26dcbfe45be243253#diff-5fdba3684bc1b07de5f0af00c27307de3a02913f4845af865164fe8a4a44e1b8R89

And the corresponding style rules here: https://github.com/benwbrum/fromthepage/pull/2349/commits/03d0af7d0bd9ab080e5493f36bfe8373db465b4e#diff-35bf25d8e5e056b4db837bc7c30baedd3fc069bb8c5c91b6dd881991cdbe764bR229

**Comment by benwbrum** [comment]
This looks like it might be a problem with my asset pipeline in development.  I see your change in `panel.js` in my source branch, but the `panel-ac107330134350d63ae6d6c6ec767929ec045ff2707af690429ca23223539862.js` file loaded into my browser does not have the change.  I've cleared my `tmp/cache`, but still see the same problem.  Any advice on getting changed assets to be picked up under `rails s`?

**Comment by benwbrum** [comment]
This was definitely a problem with my development environment.  I deleted the application and re-cloned it, and the editor works fine.

Sorry for the waste of time and effort.

**Comment by kolking** [comment]
No worries! Glad it finally works for you as it should 👍 
Btw, have you tried to stop / start rails server after deleting `tmp/cache`?

**Comment by benwbrum** [comment]
This looks good under manual testing.  We still need to figure out why tests aren't passing.

File: app/views/transcribe/display_page.html.slim [hunk 1]
```diff
@@ -45,11 +45,11 @@
 
       .toolbar_group
         -unless @preview_xml
-          =>button_tag t('.preview'), :name => 'preview', type: 'button'
+          =>button_tag t('.preview'), :name => 'preview', type: 'submit'
```
Comment (kolking) [inline app/views/transcribe/display_page.html.slim:48]: I'm not sure that's correct. A `type=submit` button will submit the form by default, but we don't want a Preview or Autolink buttons to submit the form, probably only the Save Changes button should have the `submit` type.

**Comment by benwbrum** [comment]
@kolking Historically, the preview and autolink functionalities submitted the form, since FromThePage is almost entirely server-side rendered.  

The `save_transcription` action in the controller looks for the button names and does the appropriate (non-save) things:
https://github.com/benwbrum/fromthepage/blob/main/app/controllers/transcribe_controller.rb#L198-L210

At some point we may want to switch this functionality to an AJAX flow, but that is in the future.

**Comment by kolking** [comment]
Got it, makes sense then 👍

**Pull request merged**